### ISSUE TITLE
Unable to use node-serialport (for node-dsmr); which trustedDepencies to add?

### ISSUE THREAD

**Issue by pro-sumer** [oven-sh/bun#5625]
### What version of Bun is running?

1.0.2+37edd5a6e389265738e89265bcbdf2999cb81a49

### What platform is your computer?

Linux 6.1.21-v8+ aarch64 unknown

### What steps can reproduce the bug?

Try to install [com.p1.smartmeter-dsmr.reader](https://github.com/koktaildotcom/com.p1.smartmeter-dsmr.reader) (which requires [node-dsmr](https://github.com/reneklootwijk/node-dsmr), which requires [node-serialport](https://github.com/serialport/node-serialport)) using `bun install`.

### What is the expected behavior?

* All packages/dependencies install without error
* Script/Service starts without errors

### What do you see instead?

Script/service fails with this error:

```
Sep 17 15:52:51 pi-sm p1.js[910939]:  → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/lib/binding/node-v108-linux-arm64/bindings.node
Sep 17 15:52:51 pi-sm p1.js[910939]:     at bindings (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/bindings/bindings.js:126:9)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Object.<anonymous> (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/lib/linux.js:2:36)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module._compile (node:internal/modules/cjs/loader:1256:14)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module.load (node:internal/modules/cjs/loader:1119:32)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module._load (node:internal/modules/cjs/loader:960:12)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module.require (node:internal/modules/cjs/loader:1143:19)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at require (node:internal/modules/cjs/helpers:121:18)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Object.<anonymous> (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/lib/index.js:14:22)
Sep 17 15:52:51 pi-sm p1.js[910939]:     at Module._compile (node:internal/modules/cjs/loader:1256:14) {
Sep 17 15:52:51 pi-sm p1.js[910939]:   tries: [
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/build/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/build/Debug/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/build/Release/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/out/Debug/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/Debug/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/out/Release/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/Release/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/build/default/bindings.node',
Sep 17 15:52:51 pi-sm systemd[1]: p1.service: Main process exited, code=exited, status=1/FAILURE
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/compiled/18.17.1/linux/arm64/bindings.node',
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/addon-build/release/install-root/bindings.no>
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/addon-build/debug/install-root/bindings.node>
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/addon-build/default/install-root/bindings.no>
Sep 17 15:52:51 pi-sm p1.js[910939]:     '/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/node_modules/@serialport/bindings/lib/binding/node-v108-linux-arm64/bindings.n>
Sep 17 15:52:51 pi-sm p1.js[910939]:   ]
```

### Additional information

The `node-serialport` package has lifecycle scripts that need to be [trusted](https://bun.sh/docs/cli/install#trusted-dependencies), but I don't know which package/syntax is required at the top level to achieve this.

Could also be related to #5357?

When using either `npm` or `pnpm` the lifecycle scripts are executed and I can actually use these packages, but I'd rather use `bun`!

**Comment by pro-sumer** [comment]
Looks like I need to trust `@serialport/bindings` and manually add `node-gyp`?

**Comment by pro-sumer** [comment]
Note: the above "fixes" make it possible to use Bun as package manager and NodeJS as runtime.

Using Bun as runtime as well will still fail:

```
➜ bun p1.js
121 |         throw e;
122 |       }
123 |     }
124 |   }
125 |
126 |   err = new Error(
            ^
error: Could not locate the bindings file. Tried:
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/build/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/build/Debug/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/build/Release/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/out/Debug/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/Debug/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/out/Release/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/Release/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/build/default/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/compiled/20.8.0/linux/arm64/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/addon-build/release/install-root/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/addon-build/debug/install-root/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/addon-build/default/install-root/bindings.node
 → /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/lib/binding/node-v115-linux-arm64/bindings.node
      at bindings (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/bindings/bindings.js:126:8)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/linux.js:2:6
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/linux.js:110:17)
      at overridableRequire (:1:20)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/index.js:14:11
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/@serialport/bindings/lib/index.js:14:11)
      at overridableRequire (:1:20)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/lib/index.js:2:6
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/serialport/lib/index.js:15:17)
      at overridableRequire (:1:20)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/node-dsmr/lib/connection.js:5:6
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/node-dsmr/lib/connection.js:89:0)
      at overridableRequire (:1:20)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/node-dsmr/lib/index.js:4:6
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/node_modules/node-dsmr/lib/index.js:131:0)
      at overridableRequire (:1:20)
      at /usr/local/src/com.p1.smartmeter-dsmr.reader/src/p1.js:1:6
      at globalThis (/usr/local/src/com.p1.smartmeter-dsmr.reader/src/p1.js:179:16)
      ```

**Comment by Electroid** [comment]
* We'll be fixing postinstall in this issue: #4959 
* We'll make sure you know which dependency to trust with: #5174

**Comment by Jarred-Sumner** [comment]
This particular error being thrown is fixed by @dylan-conway and @paperdave in #7132

This will be part of the Bun v1.0.17 release, which ships in a few hours from the time of writing.

I do not think `node-serialport` will work after that, however because IIRC, it relies on some libuv functions which we do not implement as of yet (Bun does not use libuv on Linux or macOS).
